injection/sharedmain/main.go

```diff
@@ -140,13 +143,23 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 	defer flush(logger)
 	ctx = logging.WithLogger(ctx, logger)
 
+	// Obtain K8s clientset.
 	kc := kubeclient.Get(ctx)
 	if err := version.CheckMinimumVersion(kc.Discovery()); err != nil {
 		logger.Fatalw("Version check failed", zap.Error(err))
 	}
 
+	// Create ConfigMaps watcher with optional label-based filter.
+	var cmLabelReqs []labels.Requirement
+	if cmLabel := system.ResourceLabel(); cmLabel != "" {
+		req, err := configmap.FilterConfigByLabelExists(cmLabel)
+		if err != nil {
+			logger.With(zap.Error(err)).Fatalf("Failed to generate requirement for label %q")
+		}
+		cmLabelReqs = append(cmLabelReqs, *req)
+	}
 	// TODO(mattmoor): This should itself take a context and be injection-based.
-	cmw := configmap.NewInformedWatcher(kc, system.Namespace())
+	cmw := configmap.NewInformedWatcher(kc, system.Namespace(), cmLabelReqs...)
```

Comment:
`cmLabelReqs...` will simply be `nil` if the `SYSTEM_RESOURCE_LABEL` env var is not set, and no filter will be applied (current behavior).